src/app.rs

```diff
@@ -366,7 +375,8 @@ impl<'a> App<'a> {
             let query_string = query_lines.clone().join(" \n");
             if !query_string.is_empty() {
               self.add_to_history(query_lines.clone());
-              let first_query = database::get_first_query(query_string.clone());
+              let dialect = get_dialect(<DB as sqlx::Database>::NAME);
```

Comment:
i think this should be moved to `main.rs`, run once, and stored as part of `app_state`